### PR TITLE
Export validated labels.npz

### PR DESCRIPTION
**Pull request by martham93**
This PR addresses issue #34 and allows users to export labels.npz files for validated classification models, and adds a checkbox to indicate if the model is binary in the UI. It adds the following columns to the database: inference type, inference list, and binary. 

cc @ingalls

### CODE REVIEW

File: ml_enabler/api/ml.py [hunk 1]
```diff
@@ -386,21 +388,71 @@ class PredictionExport(Resource):
     def get(self, model_id, prediction_id):
         req_format = request.args.get('format', 'geojson')
         req_inferences = request.args.get('inferences', 'all')
-        req_threshold = request.args.get('threshold', '0')
```
Comment (ingalls) [inline ml_enabler/api/ml.py:389]: Why did this change? This should be 0 by default, most of the time the user will manually specify a non-zero value.

File: ml_enabler/api/ml.py [hunk 2]
```diff
@@ -411,22 +463,23 @@ def generate():
...
                 if req_inferences != 'all' and row[3].get(req_inferences) is None:
                     continue
 
                 if req_inferences != 'all' and row[3].get(req_inferences) <= req_threshold:
                     continue
 
                 if req_format == "geojson" or req_format == "geojsonld":
+                    properties_dict = {}
+                    valid_dict = {}
+                    properties_dict = row[3].update(valid_dict)
```
Comment (ingalls) [inline ml_enabler/api/ml.py]: Can you post back here with an example of what the output looks like?

What I had envisioned is something like:
```JSON
"properties": {
    "school": true,
    "not school": false,
    "validity": {
        "school" true,
        "not_school": true
    }
}
```

Open to alternatives, just want to make sure I have a mental model of what this looks like

File: ml_enabler/api/ml.py [hunk 3]
```diff
@@ -411,22 +463,23 @@ def generate():
...
                     feat = {
                         "id": row[0],
                         "quadkey": row[1],
                         "type": "Feature",
-                        "properties": row[3],
+                        "properties": properties_dict,
```
Comment (ingalls) [inline ml_enabler/api/ml.py:485]: Also should we include validity information in CSV export?

File: ml_enabler/api/ml.py [hunk 4]
```diff
@@ -443,28 +496,51 @@ def generate():
...
+                headers = {
+                    "Content-Disposition": 'attachment; filename="export."' + req_format
+                }
+            )        
+            except NoValid: 
+                return {
+                    "status": 400,
+                    "error": "Can only return npz if predictions are validated. Currently there are no valid predictions"
+                }, 400
+            return Response(
```
Comment (ingalls) [inline ml_enabler/api/ml.py]: Is this accidentally duplicated?

**Review by ingalls — COMMENTED** [review]
@martham93 this is incredibly exciting. Left a couple comments to work through. Next step will be testing this further on staging and ensuring stack creation pulls from the new prediction based values

File: ml_enabler/api/ml.py (hunk 4 above)
Comment (martham93) [reply ml_enabler/api/ml.py]: yes, good catch! it's removed now

File: ml_enabler/api/ml.py (hunk 2 above)
Comment (martham93) [reply ml_enabler/api/ml.py]: I think this now aligns if there's a validated prediction the properties look like: 

` "properties": {"school": 0.651447117, "not_school": 0.258075058, "validity": {"school": false}}`

Comment (ingalls) [reply ml_enabler/api/ml.py]: :+1:

**Pull request merged**